scripts/host-testbench.sh

```diff
@@ -57,63 +57,114 @@ head -c ${INPUT_FILE_SIZE} < /dev/zero > zeros_in.raw
+if ./process_test.sh volume 32 32 48000 0; then
+  echo "volume test passed!"
+else
+  echo "volume test failed!"
+  exit 1
+fi
+
+# test with eq-iir
 echo "=========================================================="
-echo "test src with ./src_run.sh 32 32 44100 48000 zeros_in.raw src_out.raw"
-if ./src_run.sh 32 32 44100 48000 zeros_in.raw src_out.raw &>src.log; then
-  echo "src test passed!"
+echo "test eq-iir with 16 16 48000 0"
+if ./process_test.sh eq-iir 16 16 48000 0; then
+  echo "eq-iir test passed!"
 else
-  echo "src test failed!"
-  cat src.log
+  echo "eq-iir test failed!"
+  exit 1
+fi
+
+echo "test eq-iir with 24 24 48000 0"
+if ./process_test.sh eq-iir 24 24 48000 0; then
+  echo "eq-iir test passed!"
+else
+  echo "eq-iir test failed!"
+  exit 1
+fi
+
+echo "test eq-iir with 32 32 48000 0"
+if ./process_test.sh eq-iir 32 32 48000 0; then
+  echo "eq-iir test passed!"
+else
+  echo "eq-iir test failed!"
   exit 1
 fi
 
-if comparesize "$(srcsize ${INPUT_FILE_SIZE} 44100 48000)" "$(filesize src_out.raw)";then
-  echo "src_out size check passed!"
+# test with dcblock
+echo "=========================================================="
+echo "test dcblock with 32 32 48000 0"
+if ./process_test.sh dcblock 32 32 48000 0; then
+  echo "dcblock test passed!"
+else
+  echo "dcblock test failed!"
+  exit 1
+fi
+
+# test with drc
+echo "=========================================================="
+echo "test drc with 32 32 48000 0"
+if ./process_test.sh drc 32 32 48000 0; then
+  echo "drc test passed!"
 else
-  echo "src_out size check failed!"
-  cat src.log
+  echo "drc test failed!"
   exit 1
 fi
-rm src_out.raw src.log
 
-# test with eq
+# test with eq-fir
 echo "=========================================================="
-echo "test eqiir with ./eqiir_run.sh 16 16 48000 zeros_in.raw eqiir_out.raw"
-if ./eqiir_run.sh 16 16 48000 zeros_in.raw eqiir_out.raw &>eqiir.log; then
-  echo "eqiir test passed!"
+echo "test eq-fir with 32 32 48000 0"
+if ./process_test.sh eq-fir 32 32 48000 0; then
+  echo "eq-fir test passed!"
 else
-  echo "eqiir test failed!"
-  cat eqiir.log
+  echo "eq-fir test failed!"
   exit 1
 fi
 
+# test with multiband-drc
+echo "=========================================================="
+echo "test multiband-drc with 32 32 48000 0"
+if ./process_test.sh multiban-drc 32 32 48000 0; then
+  echo "multiband-drc test passed!"
+else
+  echo "multiband-drc test failed!"
+  exit 1
+fi
 
-if comparesize $INPUT_FILE_SIZE "$(filesize eqiir_out.raw)"; then
-  echo "eqiir_out size check passed!"
+# test with src
+echo "=========================================================="
+echo "test src with 32 24 48000 0"
+if ./process_test.sh src 32 24 48000 0; then
+  echo "src test passed!"
+else
+  echo "src test failed!"
+  exit 1
+fi
+
+# test with tdfb
+echo "=========================================================="
+echo "test tdfb with 32 24 48000 0"
+if ./process_test.sh tdfb 32 24 48000 0; then
```

Comment:
tdfb does not have format convert capability so the test needs to be e.g. `process_test.sh tdfb 32 32 48000 0`

Reply:
ack